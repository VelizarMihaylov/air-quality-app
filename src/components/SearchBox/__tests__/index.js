import React from 'react'
import { SearchBox } from '../index'
import { create, act } from 'react-test-renderer'

describe('SearchBox', () => {
  const data = {
    results: [
      {
        country: 'GB',
        name: 'London',
        city: 'London',
        count: 158120,
        locations: 3
      },
      {
        country: 'GB',
        name: 'Manchester',
        city: 'Manchester',
        count: 48890,
        locations: 1
      },
      {
        country: 'GB',
        name: 'Leeds',
        city: 'Leeds',
        count: 51077,
        locations: 1
      }
    ]
  }
  it('should handle loading state', () => {
    const SearchBoxRender = create(<SearchBox loading />).toJSON()
    expect(SearchBoxRender).toMatchSnapshot()
  })
  it('should handle error state', () => {
    const SearchBoxRender = create(<SearchBox error />).toJSON()
    expect(SearchBoxRender).toMatchSnapshot()
  })
  it('should list all cities options when onFocus event is triggered and data prop is set', () => {
    const SearchBoxRender = create(
      <SearchBox data={data} dispatch={jest.fn()} />
    )

    const input = SearchBoxRender.root.find(
      (element) => element.type === 'input'
    )

    act(() => {
      input.props.onFocus()
    })

    expect(
      SearchBoxRender.root.find((element) => element.type === 'ul').children
        .length
    ).toBe(3)
  })

  it('should dispatch ADD_CITY action when the city list is empty and onFocus action is triggered', () => {
    const SearchBoxRender = create(
      <SearchBox data={data} dispatch={jest.fn()} />
    )

    const input = SearchBoxRender.root.find(
      (element) => element.type === 'input'
    )

    act(() => {
      input.props.onFocus()
    })

    act(() => {
      SearchBoxRender.root
        .find((element) => element.type === 'ul')
        .children[0].props.onMouseDown()
    })

    expect(SearchBoxRender.root.props.dispatch).toHaveBeenCalledWith({
      type: 'ADD_CITY',
      payload: 'London'
    })
  })
})
