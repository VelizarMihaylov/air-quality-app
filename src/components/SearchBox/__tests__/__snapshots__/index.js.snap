// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should handle error state 1`] = `
<h1>
  Oops something went wrong!
</h1>
`;

exports[`SearchBox should handle loading state 1`] = `
<div
  className="SearchBox"
>
  <form
    className="SearchBox--form"
  >
    <div
      className="SearchBox--icon__spinner"
    />
    <input
      autoComplete="off"
      className="SearchBox--input"
      disabled={true}
      name="cities"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value=""
    />
  </form>
  <ul
    className="SearchBox--datalist"
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
